python/lsst/daf/butler/tests/_examplePythonTypes.py

```diff
@@ -25,8 +25,113 @@
 formats such as FITS or HDF5.
 """
 
+__all__ = ("ListAssembler", "MetricsAssembler", "MetricsExample", "registerMetricsExample")
+
+
 import copy
-from lsst.daf.butler import CompositeAssembler
+from lsst.daf.butler import CompositeAssembler, StorageClass
+
+
+def registerMetricsExample(butler):
```

Comment:
I assumed that the existing config files made other, possibly more complex, changes relative to the default config (the sheer number of files is a bit :dizzy_face:).

I'm already using this on [DM-22599](https://github.com/lsst/pipe_base/blob/tickets/DM-22599/tests/test_pipelineTaskTests.py), and being able to use `MetricsExample` instead of `numpy.Array` as a dummy dataset makes the test assertions much easier to read (at the cost of more boilerplate overall). I'll admit, though, that I don't expect it to be of much use in testing real `PipelineTask` classes, as there you could freely use the real data types. If you don't think it's useful here, I could move it to `pipe_base/tests/`.

Reply:
If this is useful to you then leave it here. I was more wondering whether I could make use of it in test_butler.py but I don't require that to happen on this ticket.